mkosi/__init__.py

```diff
@@ -2157,41 +2157,68 @@ def calculate_signature(context: Context) -> None:
     if not context.config.sign or not context.config.checksum:
         return
 
-    cmdline: list[PathString] = ["gpg", "--detach-sign", "--pinentry-mode", "loopback"]
+    tool = context.config.openpgp_tool
+    if tool is None or tool == "gpg":
```

Comment:
The two branches should probably be their own functions `sign_gpg` and `sign_stateless_pgp`.

This will also make the diff somewhat smaller, since the gpg code will only move and not change indentation level.